### PR TITLE
Don't clear state if you're gonna show a popup

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

medic/medic#5445

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/e2e/reports-subject.js [hunk 1]
```diff
@@ -250,7 +250,7 @@ describe('Reports Summary', () => {
     return browser
       .wait(() => element(
         by.cssContainingText('#reports-content .item-summary .sender .phone', CAROL.phone)).isPresent(),
-        10000
+        20000
```
Comment (dianabarsan) [inline tests/e2e/reports-subject.js]: it was driving me nuts.

File: webapp/src/js/controllers/inbox.js [hunk 2]
```diff
@@ -199,17 +199,18 @@ const LAST_REPLICATED_SEQ_KEY = require('../bootstrapper/purger').LAST_REPLICATE
...
-      }
-      if (toState.name.indexOf('tasks.detail') === -1) {
-        Enketo.unload($scope.form);
-        $scope.unsetSelected();
-      }
-      if (toState.name.split('.')[0] !== fromState.name.split('.')[0]){
-        $scope.$broadcast('ClearSelected');
-      }
       if (!$scope.enketoStatus.edited){
+        if (toState.name.indexOf('reports') === -1 || toState.name.indexOf('contacts') === -1 || toState.name.indexOf('tasks') === -1 || toState.name.indexOf('messages.detail') === -1) {
```
Comment (dianabarsan) [inline webapp/src/js/controllers/inbox.js]: I resisted the urge to remove these if clauses, even if they're doing exactly nothing. 
- first clause is always true, unless your to state is `reports.contacts.tasks.messages.detail`
- second clause is useless cause only the child scope has a `form` property and `$scope.unsetSelected()` was already called in the first clause
- `$scope.unsetSelected()` also broadcasts `ClearSelected`

**Comment by dianabarsan** [comment]
Please review @garethbowen . Thank you!

File: tests/e2e/reports-subject.js (hunk 1 above)
Comment (garethbowen) [reply tests/e2e/reports-subject.js]: I wonder if that'll help...

File: webapp/src/js/controllers/inbox.js (hunk 2 above)
Comment (garethbowen) [reply webapp/src/js/controllers/inbox.js]: The third one may be useful because it'll fire at different times. It should be an `else if` though.

**Pull request merged**